### DISCUSSION TITLE
Access bridge network from macvlan network

### DISCUSSION THREAD

**Discussion by MajorTomDE** [wg-easy/wg-easy#358] [Q&A]
Hello folks,

this is not a bug, else more a how to question:

I'm running wg-easy within a docker installation on my Raspberry PI

wg-easy runs as macvlan with IP 192.168.178.203
heimdall and pi-hole runs as bridge network with IP 192.168.178.2

I have the following challange:
When I activate the wireguard connection to my network, I'm able to open the wg-easy interface (192.168.178.203) or access my FritzBox Router (192.168.178.1)

But I'm not able to open the heimdall or pi-hole interface on 192.168.178.2

It seems, per default it is not possible to access a container running in bridge network from an container running in macvlan an vice verca.

Do you have an idea how I can resolve this?

Thank you
Thomas

**Comment by Matthi-17** [comment]
Hey MajorTomDE,

you are right, you can not access a macvlan container from a afer container. It is a feature, when you google you find something why.

For my case, I have this problem with ioBroker. I added a new bridge network to my iobroker an my influxdb, and than they can comunicate.

I hope I can help you. Falls nicht schreib mir ruhig.

**Comment by Matthi-17** [comment]
Hey Thomas,

one other thing.

Are you using docker-compose? I have the problem that I can not access my complett home subnet behindet the macvlan off my container. Can you share your config?

Thanks
Matthias

**Comment by Matthi-17** [comment]
Hello Thomas,

so I fixed my problem. And  will tell you how I  fixed yours for me.

First I have an nginx proxy manager. All my FQDN I will access over vpn, which are docker containers, are in a same bridge network then the nginx. And the nginx also have a mac vlan adress, so he can manage the comunication for me.

I hope it help you and other people :)

Have a nice time all together

Matthias

**Comment by xrayos** [comment]
has anyone figured out how to fix this?

synology docker, running pihole, unbound, and wireguard

wireguard connection works. I can connect to every device on LAN, except the containers on macvlan (ie, pihole)

thx

**Comment by Matthi-17** [comment]
Hi,

you have found a Feature of Docker I know yet. When you have a Docker Container on MACVLAN this container can not communicate with the bridge Container by default. 

For me I fixed this with an Nginx Proxymanager who communicated into the Bridgenetworks 

I hope it helps. 



> has anyone figured out how to fix this?
> 
> 
> 
> synology docker, running pihole, unbound, and wireguard
> 
> 
> 
> wireguard connection works. I can connect to every device on LAN, except the containers on macvlan (ie, pihole)
> 
> 
> 
> thx
